test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs

```diff
@@ -61,6 +61,124 @@ public void LightGBMBinaryEstimator()
             Done();
         }
 
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // LightGBM is 64-bit only
+        public void LightGBMBinaryEstimatorMonotoneThroughOptions()
+        {
+            var (pipe, dataView) = GetBinaryClassificationPipeline();
+
+            var trainer = ML.BinaryClassification.Trainers.LightGbm(new Options
+            {
+                NumLeaves = 10,
+                NThread = 1,
+                MinDataPerLeaf = 2,
+                MonotoneConstraints = new string[] {"pos:1-3","neg:4" }
+            });
+
+            var pipeWithTrainer = pipe.Append(trainer);
+            TestEstimatorCore(pipeWithTrainer, dataView);
+
+            var transformedDataView = pipe.Fit(dataView).Transform(dataView);
+            var model = trainer.Train(transformedDataView, transformedDataView);
+            Done();
```

Comment:
Could you check if prediction values are really monotone? For example, feeding 10 examples such as [0, 0, 1, 0], [0, 0, 2, 0], [0, 0, 3, 0], ... into the trained model and then examine their prediction values.